tests/integration_tests/functional/test_swiotlb.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Test swiotlb related functionality."""
+import re
+
+import pytest
+
+from framework.properties import global_props
+
+pytestmark = pytest.mark.skipif(
+    global_props.cpu_architecture != "aarch64", reason="swiotlb only supported on ARM"
+)
+
+
+def test_swiotlb_boot(microvm_factory, guest_kernel_linux_6_1, rootfs):
+    """Tests that a VM can boot if all virtio devices are bound to a swiotlb region, and
+    that this swiotlb region is actually discovered by the guest."""
+    vm = microvm_factory.build(guest_kernel_linux_6_1, rootfs)
+    vm.spawn()
+    vm.basic_config(memory_config={"initial_swiotlb_size": 64})
+    vm.add_net_iface()
+    vm.start()
+
+    _, dmesg, _ = vm.ssh.check_output("dmesg")
+
+    assert (
+        "OF: reserved mem: initialized node bouncy_boi, compatible id restricted-dma-pool"
+        in dmesg
+    )
+
+    match = re.search(r"Placed swiotlb region at \[(\d+), (\d+)\)", vm.log_data)
+
+    assert match is not None, "Firecracker did not print swiotlb region placement"
+
+    swiotlb_start, swiotlb_end = match.group(1, 2)
+
+    found_any = False
+
+    for match in re.finditer(
+        r"Placed virt queue ([a-zA-Z ]+) at \[(\d+), (\d+)\)", vm.log_data
+    ):
+        found_any = True
```

Comment:
do we want "any" or "all"?

Reply:
any. This is just to check the loop isn't a no-op

Reply:
what I mean is wouldn't it be useful to check that all devices we attach allocate queues in the swiotlb?

Reply:
yeah, that's what the assert in the loop does. This boolean is only to verify that firecracker actually printed the log message we're expecting.

Reply:
ah, see it now, thx.